pinc/ProjectState.inc

```diff
@@ -28,39 +28,48 @@ class ProjectState
 
 class ProjectStates
 {
-    private static $states = [];
+    /** @var array<string, ProjectState> */
+    private static array $states = [];
 
-    public static function add_state($state)
+    public static function add_state(ProjectState $state): void
     {
         self::$states[$state->name] = $state;
     }
 
-    public static function get_states()
+    /** @return string[] */
+    public static function get_states(): array
     {
         return array_keys(self::$states);
     }
 
-    public static function get_all()
+    /** @return array<string, ProjectState> */
+    public static function get_all(): array
     {
         return self::$states;
     }
 
-    public static function get_medium_label($state)
+    // TODO: We should ensure that `$state` is never null. For now, we allow it.
+    public static function get_medium_label(?string $state): string
```

Comment:
Some smoketest were sending null to this function (yay to smoke tests!), thus I have relaxed the string arguments I couldn't be sure of to allow the null.